atom/browser/ui/tray_icon_cocoa.mm

```diff
@@ -280,6 +286,17 @@ - (void)mouseUp:(NSEvent*)event {
   if (menuController_)
     return;
 
+  // If we are ignoring double click events, we should ignore the `clickCount`
+  // value and immediately emit a click event.
+  if (ignoreDoubleClickEvents_ == YES) {
+    trayIcon_->NotifyClicked(
```

Comment:
This shouldn't be simply copypasted.
Please rewrite it with something like that (pseudocode):

```js
shouldBeHandledAsASingleClick = (event.clickCount == 1) || ignoreDoubleClickEvents
if (shouldBeHandledAsASingleClick) {
  trayIcon_->NotifyClicked(...)
}

shouldBeHandledAsADoubleClick = (event.clickCount == 2) && !ignoreDoubleClickEvents
if (shouldBeHandledAsADoubleClick) {
  trayIcon_->NotifyDoubleClicked(...)
}

...
```

Reply:
Good idea, I like this a lot more. Updated!